app/controllers/admin/badge_achievements_controller.rb

```diff
@@ -0,0 +1,49 @@
+module Admin
+  class BadgeAchievementsController < Admin::ApplicationController
+    layout "admin"
+
+    def index
+      @q = BadgeAchievement
+        .includes(:badge)
+        .includes(:user)
+        .order(created_at: :desc)
+        .ransack(params[:q])
+      @badge_achievements = @q.result.page(params[:page] || 1).per(15)
+    end
+
+    def destroy
+      @badge_achievement = BadgeAchievement.find(params[:id])
+
+      if @badge_achievement.destroy
+        flash[:success] = "Badge has been deleted!"
+      else
+        flash[:danger] = @badge_achievement.errors_as_sentence
+      end
+      redirect_to admin_badge_achievements_path
+    end
+
+    def award
+      @badge = Badge.page(params[:page]).per(Badge.count)
```

Comment:
So, my only hesitation with changing this to a fixed number, like 15 or 20, is that I think the reason why this loads all of the badges for the Admin is so that they are able to select any badge from the badges that exist and award it to a user at any given time (_I believe_). 

As for changing the variable name here to be something more explicit, I was able to do this without an issue and went ahead and changed it to `@all_badges`. How does this sound @mstruve? 😄

Reply:
>  I think the reason why this loads all of the badges for the Admin is so that they are able to select any badge

By using `page` and `per` here you are setting up the view for pagination so I would assume the admin could page through all of the badges to find the one they are looking for. 

> I was able to do this without an issue and went ahead and changed it to @all_badges

WOOT! 🙌

Reply:
That's a fair point @mstruve! I just sanity checked myself to make sure that we needed and wanted access to all of the Badges here and it seems like that is the case. That being said, I've changed this to `@all_badges = Badge.all.select(:title, :slug)` to optimize the query a little more and removed the SQL paging since we do not need pagination here and it's adding unnecessary code to the controller. How do you feel about this?